Update cq sdk version

bbernays opened the pull request:
Implements #26 


- [x] ~~restructuring the resources to sub-packages as in other providers~~ DO doesn't have services so all resources are staying in a single package
- [x] use the new TestResource SDK function
- [x] ~~splitting tests into mock and integration tests~~ there are no mock tests
- [ ] re-writing all terraform files to use modules.

bbernays commented on docs/docs.go:
```diff
@@ -1,15 +1,15 @@
+	err := docs.GenerateDocs(provider.Provider(), "./docs", true)
 	if err != nil {
-		fmt.Sprintf("Failed to geneerate docs: %s", err)
+		log.Fatalf("Failed to generate docs: %s", err)
```
Bringing this in line with other providers

yevgenypats commented on resources/account_test.go:
```diff
@@ -0,0 +1,14 @@
+)
+
+func TestIntegrationAccount(t *testing.T) {
+	client.DOTestHelper(t, Account(), "./snapshots")
```
we are not using snapshots anymore so I think this parameter is unused (I think you can search and replace/delete).

bbernays replied:
Removed

yevgenypats commented on .github/workflows/int_test.yml:
```diff
@@ -48,7 +48,7 @@ jobs:
 
       - name: Integration Test
-        run: go test -v -p 20 ./resources/integration_tests
+        run: go test -tags=integration ./...
```
I'd add a timeout here (will test first how long it takes and will add this +1 min `-timeout=2min` or something like that)

bbernays replied:
Fixed

yevgenypats reviewed (changes requested):
Looks good. few minor comments.
The terraform files can be added in this or follow-up. I don't know if DO has a lot of modules so maybe the only thing that will be necessary is just removing the whole `${var.test_prefix}` and `${var.test_suffix}` from  the tf files and reapply them.

bbernays commented:
> Looks good. few minor comments. The terraform files can be added in this or follow-up. I don't know if DO has a lot of modules so maybe the only thing that will be necessary is just removing the whole `${var.test_prefix}` and `${var.test_suffix}` from the tf files and reapply them.

I have addressed all of the comments. I also scheduled a few min for us to discuss the tf modules tomorrow so I will handle that in a second pr
